### PR TITLE
vectorize spl

### PR DESCRIPTION
**Pull request by randrescastaneda**
Hi @tonyfujs , 

I just vectorized the `cokmpute_spl` function. Could you please merge it to dev? I need it to include the SPL in the pipeline. Thank you!
Best,

### CODE REVIEW

**Comment by randrescastaneda** [comment]
Hi @tonyfujs , 

Whenever you have a chance, could could you please merge this PR. I need it to estimate the SPL in the pipeline. 

Thank you.

File: R/compute_spl.R [hunk 1]
```diff
@@ -3,32 +3,36 @@
...
 #' @keywords internal
 compute_spl <- function(weighted_median_ppp,
-                        threshold_rate = 0.5,
-                        min_level = 1.9) {
+                        ppp_year) {
 
-  assertthat::assert_that(
-    !(threshold_rate >= 1 | threshold_rate <= 0),
-    msg = "`threshold_rate` must between 0 and 1.")
+  threshold_rate <-  0.5
```
Comment (tonyfujs) [inline R/compute_spl.R:19]: Why did you move it to the function body (instead of leaving it as a function argument)?

File: R/compute_spl.R [hunk 2]
```diff
@@ -3,32 +3,36 @@
...
+
+  if (ppp_year == 2011) {
+    constant  <- 1
+    min_level <- 1.9
+  } else if (ppp_year == 2017) {
+    constant  <- 1.15
+    min_level <- 2.15
+  } else {
+    cli::cli_abort("Invalid PPP year. {.field ppp_year} must be either 2011 or 2017")
+  }
```
Comment (tonyfujs) [inline R/compute_spl.R:29]: As discussed on the phone. Let's keep `wbpip` a relatively low-level package to power the API and an upcoming higher-level package for end-users to use directly. `cli` interactive message will be there in this interactive package.

File: R/compute_spl.R [hunk 3]
```diff
@@ -3,32 +3,36 @@
...
-    !(threshold_rate >= 1 | threshold_rate <= 0),
-    msg = "`threshold_rate` must between 0 and 1.")
+  threshold_rate <-  0.5
+
+  if (ppp_year == 2011) {
+    constant  <- 1
+    min_level <- 1.9
+  } else if (ppp_year == 2017) {
+    constant  <- 1.15
+    min_level <- 2.15
```
Comment (tonyfujs) [inline R/compute_spl.R:26]: I'm not sure this should be ahrd-coded in the function's body... This should probably be either function arguments that take their value when being called from higher level function...

**Review by tonyfujs — CHANGES_REQUESTED** [review]

**Comment by randrescastaneda** [comment]
I'll make the PR to DEV

**Pull request merged**